### ISSUE TITLE
[Accessibility]: Disabled buttons, inputs and dropdowns have insufficient color contrast in both dark and light mode

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#35734]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34195

### Problem description

Unsure if this issue is specific to the code monitoring page or if this is global. The page is https://sourcegraph.com/code-monitoring/new 

The buttons and dropdowns have insufficient color contrast when disabled. This is the case for both dark and light mode.

Screenshots:
![image](https://user-images.githubusercontent.com/206864/169374226-2fbf554d-cb48-43c2-84e6-ecd355a27fe8.png)
![image](https://user-images.githubusercontent.com/206864/169374253-f519cbdf-716d-477d-81db-def04735e209.png)
![image](https://user-images.githubusercontent.com/206864/169374280-ed4d92d9-1bd8-4d10-b18b-56c13ad55b56.png)
![image](https://user-images.githubusercontent.com/206864/169374330-f6080670-0ac6-450f-b6e7-092024f3e3e8.png)
![image](https://user-images.githubusercontent.com/206864/169374618-5743c70c-86bc-424b-a34d-207f2ed2e479.png)
![image](https://user-images.githubusercontent.com/206864/169375190-173c0e4b-f56a-40df-94e9-4adc2bfd9598.png)

### Expected behavior

Contrast checks pass.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by courier-new** [comment]
+1 to this issue. The new server-side batch changes workflow makes extensive use of disabled buttons and inputs.

**Comment by taylorsperry** [comment]
Is this related to the other contrast work that you're addressing, @umpox? I'm adding our team label just to keep track 👍

**Comment by sourcegraph-bot-2** [comment]
Heads up @taylorsperry @jasongornall - the "team/frontend-platform" label was applied to this issue.

**Comment by felixfbecker** [comment]
[1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG22/#contrast-minimum) explicitly does not apply to disabled UI elements:

> [...] except the following: [...]
> Incidental: Text or images of text that are part of an inactive [user interface component](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components)

That doesn't mean it's a very good pattern that should be used widely, but it doesn't block WCAG compliance. Increasing the contrast would make the buttons hard to distinguish from active elements, which would be another accessibility problem.

In many cases it's better to keep the button active, and then show a message when its being clicked. E.g. a submit button that triggers form validation on click pointing out which fields are missing, as opposed to being disabled until the form is completely valid.
